packages/devtools_app/lib/src/code_size/file_import_container.dart

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:path/path.dart';
+
+import '../theme.dart';
+import '../ui/label.dart';
+
+class FileImportContainer extends StatefulWidget {
+  const FileImportContainer({
+    @required this.title,
+    this.actionText,
+    this.onAction,
+    this.onFileSelected,
+    this.importNewFile = true,
+    Key key,
+  }) : super(key: key);
+
+  final String title;
+
+  /// The title of the action button.
+  final String actionText;
+
+  final Function(String filePath) onAction;
+
+  final Function(String filePath) onFileSelected;
+
+  // TODO(peterdjlee): Remove once the file picker is implemented.
+  final bool importNewFile;
+
+  @override
+  _FileImportContainerState createState() => _FileImportContainerState();
+}
+
+class _FileImportContainerState extends State<FileImportContainer> {
+  String importedFile;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Text(
+          widget.title,
+          style: const TextStyle(fontSize: 18.0),
+        ),
+        const SizedBox(height: defaultSpacing),
+        Expanded(
+          child: Container(
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(15.0),
+              color: Theme.of(context).colorScheme.chartAccentColor,
+            ),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                Column(
+                  children: [
+                    _buildImportButton(),
+                    if (importedFile != null)
+                      _buildImportedFileDisplay(context),
```

Comment:
instead of doing this, maybe display "No file selected" if imported file == null